### PR TITLE
Feature/input form field suffix prefix

### PR DESCRIPTION
**Pull request by Rafik-Belkadi**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] icon
- [x] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the new behavior?

Added ability to have prefix/suffix inside an input

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ui/input/helm/src/lib/hlm-input.directive.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { cva, VariantProps } from 'class-variance-authority';
 import { ClassValue } from 'clsx';
 
 export const inputVariants = cva(
-	'flex rounded-md border font-normal border-input bg-transparent text-sm ring-offset-background file:border-0 file:text-foreground file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
+	'flex rounded-md border font-normal border-input bg-transparent text-sm ring-offset-background file:border-0 file:text-foreground file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring  disabled:cursor-not-allowed disabled:opacity-50',
```
Comment (goetzrobin) [inline libs/ui/input/helm/src/lib/hlm-input.directive.ts:7]: I'm curious about the reason behind this change?

File: libs/ui/input/helm/src/lib/hlm-input-form-field.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+@if (hasLabel) {
+	<ng-content select="[hlmLabel]"></ng-content>
+}
+<div class="relative block {{ class }}">
+	@if (hasPrefix) {
+		<div class="absolute left-4 top-0 flex h-10 items-center justify-center">
+			<ng-content select="[hlmInputPrefix]"></ng-content>
```
Comment (goetzrobin) [inline libs/ui/input/helm/src/lib/hlm-input-form-field.component.html:7]: Let's use self closing tags everywhere please

File: libs/ui/input/helm/src/lib/hlm-input-form-field.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { HlmLabelDirective } from '@spartan-ng/ui-label-helm';
+import { HlmInputPrefixDirective } from './hlm-input-prefix.directive';
+import { HlmInputSuffixDirective } from './hlm-input-suffix.directive';
+import { HlmInputDirective } from './hlm-input.directive';
+
+@Component({
+	standalone: true,
+	selector: 'hlm-input-form-field',
+	imports: [HlmInputDirective, HlmInputSuffixDirective, HlmInputPrefixDirective, HlmLabelDirective, NgClass],
+	templateUrl: './hlm-input-form-field.component.html',
```
Comment (goetzrobin) [inline libs/ui/input/helm/src/lib/hlm-input-form-field.component.ts:12]: Let's move the template into the component directly

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Looks great! Left some comments :) excited for this 😍

File: libs/ui/input/helm/src/lib/hlm-input.directive.ts (hunk 1 above)
Comment (Rafik-Belkadi-malou) [reply libs/ui/input/helm/src/lib/hlm-input.directive.ts:7]: when focused, it was adding a "double" ring around the input, which looks really bad when it has an error, or even when focused without an error

Comment (goetzrobin) [reply libs/ui/input/helm/src/lib/hlm-input.directive.ts:7]: Can you send me a screenshot of what you see? I am wondering if this is just how it looks with shadcn or if there's something wrong. Thank you!

**Comment by goetzrobin** [comment]
@Rafik-Belkadi let me know if there's anything you need me to do to support you on this

**Comment by goetzrobin** [comment]
Closing this for now. Please reopen if you have the capacity to make the requested changes :)

**Pull request closed without merging**